controllers/component_build_controller_unit_test.go

```diff
@@ -1019,6 +1019,65 @@ func TestGeneratePACRepository(t *testing.T) {
 					Name: pipelinesAsCodeWebhooksSecretName,
 					Key:  getWebhookSecretKeyForComponent(getComponent("https://github.com/user/test-component-repository", nil)),
 				},
+				URL:  "https://github.com",
+				Type: "github",
+			},
+		},
+		{
+			name:    "should create PaC repository for GitHub application on self-hosted GitHub",
+			repoUrl: "https://github.self-hosted.com/user/test-component-repository",
+			componentAnnotations: map[string]string{
+				GitProviderAnnotationName: "github",
+				GitProviderAnnotationURL:  "https://github.self-hosted.com",
```

Comment:
Do we  have a test where the protocol is missing?

Reply:
Good catch, now we do